src/spaces/grid_general.jl

```diff
@@ -148,6 +148,24 @@ function nearby_positions(
     return (mod1.(n .+ pos, space_size) for n in nindices)
 end
 
+function random_nearby_position(pos::ValidPos, model::ABM{<:AbstractGridSpace{D,false}}, r=1; kwargs...) where {D}
+    nindices = offsets_within_radius_no_0(model.space, r)
+    stored_ids = model.space.stored_ids
+    rng = abmrng(model)
+    while true
+        chosen_offset = rand(rng, nindices)
+        chosen_pos = pos .+ chosen_offset
+        checkbounds(Bool, stored_ids, chosen_pos...) && return chosen_pos
+    end
+end
+
+function random_nearby_position(pos::ValidPos, model::ABM{<:AbstractGridSpace{D,true}}, r=1; kwargs...) where {D}
+    nindices = offsets_within_radius_no_0(model.space, r)
+    chosen_offset = rand(abmrng(model), nindices)
+    chosen_pos = mod1.(pos .+ chosen_offset, spacesize(model))
```

Comment:
Isn't it faster here to check for bounds, and only if out of bounds call the `mod1` function? I think you showed that in another PR.

Reply:
right, I actually think that there are some more places where this applies but for now let's solve just the issue just for this case :-)